backend/tests/src/lib/services/auth/auth.service.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  ApiPath,
+  AuthApiPath,
+  UserSignUpRequestDto,
+  UserSignUpResponseDto,
+} from 'guruhub-shared';
+
+import { Response } from '~/lib/common/types/types';
+
+import { HttpService } from '../http/http.service';
+
+type Constructor = {
+  httpService: HttpService;
+};
+
+class AuthService {
+  #httpService: HttpService;
+
+  public constructor({ httpService }: Constructor) {
+    this.#httpService = httpService;
+  }
+
+  public signUp(
+    data: UserSignUpRequestDto,
+  ): Promise<Response<UserSignUpResponseDto>> {
+    return this.#httpService
```

Comment:
Must be "return await this.#httpService", because asynchronous operation

Reply:
No. `.send` returns a promise and this promise is returned from `.signUp`. Even if `.signUp` will be `async`, everything will work. JS is clever :)

Reply:
And linter sometimes doesn't allow `return await` at all